server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -168,8 +171,14 @@ public InputStream downloadBlob(Iterable<String> path, String fileName) throws I
 
     @Override
     @ExperimentalApi
-    public FetchBlobResult downloadBlobWithMetadata(Iterable<String> path, String fileName) throws IOException {
-        return blobStore.blobContainer((BlobPath) path).readBlobWithMetadata(fileName);
+    public InputStreamWithMetadata downloadBlobWithMetadata(Iterable<String> path, String fileName) throws IOException {
+        // If the blob store supports metadata, retrieve the blob with metadata; otherwise, retrieve the blob without metadata.
+        if (blobStore.isBlobMetadataSupported()) {
+            return blobStore.blobContainer((BlobPath) path).readBlobWithMetadata(fileName);
```

Comment:
No, we should not return `readBlobWithMetadata` always. readBlobWithMetadata is not implemented for every type of blob stores. That is the reason we kept it with a name indicating that this will return blob inputstream with metadata  as well.